postgres-driver/relay.go

```diff
@@ -3,104 +3,139 @@ package postgresdriver
+	$13::varchar[],
+	$14::varchar[],
+	$15::error_sources_enum[],
+	$16::varchar[],
+	$17::integer[],
+	$18::varchar[],
+	$19::integer[],
+	$20::integer[],
+	$21::integer[],
+	$22::boolean[],
+	$23::integer[],
+	$24::boolean[],
+	$25::boolean[],
+	$26::varchar[],
+	$27::varchar[],
+	$28::date[],
+	$29::date[]
   ) AS t(
-	chain_id,
+	relay_id,
+	pokt_chain_id,
 	endpoint_id,
 	session_key,
+	relay_source_url,
 	pokt_node_address,
+	pokt_node_domain,
+	pokt_node_public_key,
 	relay_start_datetime,
 	relay_return_datetime,
 	is_error,
 	error_code,
 	error_name,
 	error_message,
+	error_source,
 	error_type,
 	relay_roundtrip_time,
-	relay_chain_method_id,
+	relay_chain_method_ids,
 	relay_data_size,
 	relay_portal_trip_time,
 	relay_node_trip_time,
 	relay_url_is_public_endpoint,
 	portal_origin_region_id,
 	is_altruist_relay,
+	is_user_relay,
+	request_id,
+	pokt_tx_id,
 	created_at,
 	updated_at
   )`
 
+const chainMethodIDSeparator = ","
+
 func (d *PostgresDriver) WriteRelay(ctx context.Context, relay types.Relay) error {
 	now := time.Now()
 
 	return d.InsertRelay(ctx, InsertRelayParams{
-		ChainID:                  relay.ChainID,
+		RelayID:                  relay.RelayID,
+		PoktChainID:              relay.PoktChainID,
 		EndpointID:               relay.EndpointID,
 		SessionKey:               relay.SessionKey,
+		RelaySourceUrl:           relay.RelaySourceURL,
 		PoktNodeAddress:          relay.PoktNodeAddress,
+		PoktNodeDomain:           relay.PoktNodeDomain,
+		PoktNodePublicKey:        relay.PoktNodePublicKey,
 		RelayStartDatetime:       relay.RelayStartDatetime,
 		RelayReturnDatetime:      relay.RelayReturnDatetime,
 		IsError:                  relay.IsError,
-		ErrorCode:                newSQLNullInt32(relay.ErrorCode),
+		ErrorCode:                newSQLNullInt32(int32(relay.ErrorCode)),
 		ErrorName:                newSQLNullString(relay.ErrorName),
 		ErrorMessage:             newSQLNullString(relay.ErrorMessage),
-		ErrorType:                newSQLNullErrorType(ErrorTypesEnum(relay.ErrorType)),
-		RelayRoundtripTime:       relay.RelayRoundtripTime,
-		RelayChainMethodID:       relay.RelayChainMethodID,
-		RelayDataSize:            relay.RelayDataSize,
-		RelayPortalTripTime:      relay.RelayPortalTripTime,
-		RelayNodeTripTime:        relay.RelayNodeTripTime,
+		ErrorType:                newSQLNullString(relay.ErrorType),
+		ErrorSource:              newSQLNullErrorSource(ErrorSourcesEnum(relay.ErrorSource)),
+		RelayRoundtripTime:       int32(relay.RelayRoundtripTime),
+		RelayChainMethodIds:      strings.Join(relay.RelayChainMethodIDs, chainMethodIDSeparator),
+		RelayDataSize:            int32(relay.RelayDataSize),
+		RelayPortalTripTime:      int32(relay.RelayPortalTripTime),
+		RelayNodeTripTime:        int32(relay.RelayNodeTripTime),
 		RelayUrlIsPublicEndpoint: relay.RelayURLIsPublicEndpoint,
-		PortalOriginRegionID:     relay.PortalOriginRegionID,
+		PortalOriginRegionID:     int32(relay.PortalOriginRegionID),
```

Comment:
`portalOriginRegionID` is based on the `PortalRegion` table but tbh with `PortalRegion` having so few values (literally less than 10) I think the primary key could be changed to the region name instead of an ID, lmk what you think